yaml_edit/lib/src/map_mutations.dart

```diff
@@ -47,14 +44,9 @@ SourceEdit updateInMap(
 
 /// Performs the string operation on [yaml] to achieve the effect of removing
 /// the element at [key] when re-parsed.
-SourceEdit removeInMap(YamlEditor yamlEdit, YamlMap map, Object key) {
-  ArgumentError.checkNotNull(yamlEdit, 'yamlEdit');
-  ArgumentError.checkNotNull(map, 'map');
-
-  if (!containsKey(map, key)) return null;
```

Comment:
`removeInMap()` is only used in `remove()` where the key is guaranteed. I've removed the `containsKey()` call so the return value can be non-nullable.

Reply:
We could add an `assert(containsKey(map, key))` for extra sanity if we wanted to...